### ISSUE TITLE
Azure VXC Destroy doesn't work

### ISSUE THREAD

**Issue by alexeyvromanov** [megaport/terraform-provider-megaport#43]
Something changed on Azure side and destroy for VXC doesn't work anymore and requires to remove peering connection from express route first, probably need some adjustment for VXC deletion process on Megaport side

│ Error: Error deleting resource f026542c-aef0-4a8c-b7d4-56b0b9fbceca: has an attached peering connection that must be removed in Azure first:

**Comment by mega-alex** [comment]
This has been reworked in the `v1.0.0` provider. The underlying issue is that the peering can't be deleted by megaport, only created. You can specify the peering using the azure terraform provider, [here](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/express_route_circuit_peering) is the documentation for that. The azure provider can tear that down first before the VXC is deleted.

**Comment by mega-alex** [comment]
I'll do some more testing on this to confirm, but I think this may be a timing issue with the Azure APIs we use internally not updating at the same time as the public API when you delete the ExpressRoute. 

I would try to delete the express route in a different terraform apply, wait for a bit, and then try to delete the VXC to see if this issue is fixed.

**Comment by rybakovanton-metta** [comment]
thank you!
I forgot to mention this vxc connected to MVE-Cisco (if it's matter)

**Comment by mega-alex** [comment]
Does deleting the ExpressRoute first, then waiting for a few minutes, then tearing down the VXC work?

**Comment by rybakovanton-metta** [comment]
Yes, if I delete it via portal wait a little and start destroy again, it's working.
